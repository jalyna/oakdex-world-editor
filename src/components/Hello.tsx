import * as React from 'react'
import styled from 'styled-components'

export interface HelloProps { 
  compiler: string
  framework: string
}


// 'HelloProps' describes the shape of props.
// State is never set so we use the '{}' type.
export default class Hello extends React.Component<HelloProps, {}> {
  render() {
    return <h1><Button>Helloe</Button> from {this.props.compiler} and {this.props.framework}!</h1>
  }
}

const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 1em;
  padding: 0.25em 1em;
`
