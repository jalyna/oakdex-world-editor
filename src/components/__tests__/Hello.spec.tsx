import * as React from 'react'
import { shallow } from 'enzyme'
import Hello from '../Hello'

describe('Hello', () => {
  it('renders', () => {
    const hello = shallow(<Hello compiler='foo' framework='lala' />)
    expect(hello).toMatchSnapshot()
  })
})
