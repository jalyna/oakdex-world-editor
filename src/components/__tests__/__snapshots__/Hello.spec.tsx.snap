// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders 1`] = `
<h1>
  <styled.button>
    Helloe
  </styled.button>
   from 
  foo
   and 
  lala
  !
</h1>
`;
